### PR TITLE
Python3 support

### PR DESCRIPTION
**Pull request by procrastinatio**
As all tentatives to ditch or declutter `mf-chsdi3` have failed in the past, and will likely also fail in the near future, we have to support `python3` to secure the operation for the next few years.

* Code is compatible `python2` and `python3>=3.6`
* Use of Python 3.6. Python 3.7 need much more work to run on Debian Jessie
* All `integrations` and `functionals` tests are OK. 
* About 10 `e2e` tests are failing against `test`, but none against `prod` (missing some WMS layers).
* The service `search` is based on the very dated `sphinxapi`. We basically have a fork here.
* Additional work: deploy scripts and upgrade to Debian **buster**

[Demo with python2](https://mf-chsdi3.int.bgdi.ch/goetterdaemmerung) (should still work)

**Deploy/apache**

* Porting `deploy` utility and various scripts
* Main `wsgi` script (or ditching `mod_wsgi`?)

**Functionals tests**
* [x] test_catalog_tree.py
* [x] test_clientdata_dynamodb.py
* [x] test_helpers.py
* [x] test_mortonspacekey.py
* [x] test_parser.py
* [x] test_renderer.py
* [x] test_types.py
* [x] test_vector.py

**Integration tests**

* [x] test_admin.py
* [x] test_cache_update.py
* [x] test_catalog.py
* [x] test_checker.py
* [x] test_color.py  [OK]
* [x] test_features_service.py  
* [x] test_file_storage.py    OK
* [x] test_glstyles_storage.py   OK
* [x] test_historicalmaps.py
* [x] test_identify_service.py  OK
* [x] test_layers.py   OK
* [x] test_layers_service.py OK
* [x] test_loader.py
* [x] test_luftbilder.py
* [x] test_opentransapi.py
* [x] test_qrcode.py
* [x] test_search.py    OK!!!!
* [x] test_shortener.py
* [x] test_sitemap.py   OK
* [x] test_sphinxapi.py  OK
* [x] test_stationboard.py  OK (SKIP=6)
* [x] test_topics.py
* [x] test_translation.py
* [x] test_wmtscapabilities.py  OK

**End-to-end tests**

* [x] test_links.py         OK
* [x] test_varnish.py    OK
* [x] test_wms.py         OK (prod)
* [x] test_wmtscapabilities_auth.py OK
* [x] test_wmtsgettile.py  OK (prod)
* [x] test_wmts.py    OK

*Details for test_search.py* --> All OK

### CODE REVIEW

**Comment by pakb** [comment]
image

File: Makefile [hunk 1]
```diff
@@ -148,6 +174,8 @@ help:
 	@echo "- cleanall           Remove all the build artefacts"
 	@echo
 	@echo "Variables:"
+	@echo "PYTHON_VERSION:      ${PYTHON_VERSION}"
+	@echo "PYTHON_CMD: i        ${PYTHON_CMD}"
```
Comment (rebert) [inline Makefile]: i from 'vi insert?'

**Comment by procrastinatio** [comment]
Current status, after 29 commits
```
Ran 100 tests in 50.855s

FAILED (SKIP=2, errors=5, failures=9)

```

**Comment by procrastinatio** [comment]
```
Ran 3579 tests in 493.537s

FAILED (SKIP=12, errors=746, failures=29)
Makefile:232: recipe for target 'test' failed
make: *** [test] Error 1

```

**Comment by procrastinatio** [comment]
Looks doable
```
Ran 3580 tests in 545.712s

FAILED (SKIP=12, errors=17, failures=25)
Makefile:232: recipe for target 'test' failed
``

**Comment by procrastinatio** [comment]
No more errors, only failures in both `integrations` and `e2e` tests.

```
Ran 3580 tests in 622.550s
FAILED (SKIP=12, failures=32)
Makefile:232: recipe for target 'test' failed


Ran 3936 tests in 623.504s
FAILED (failures=21)
Makefile:236: recipe for target 'teste2e' failed
```

**Comment by procrastinatio** [comment]
`functionals` and `integrations`tests are OK (or skipping)
```
Ran 3604 tests in 547.665s

OK (SKIP=12)

```

**Comment by procrastinatio** [comment]
`e2etests` 

```
Ran 3973 tests in 706.838s

FAILED (failures=11)

```

File: tests/integration/test_glstyles_storage.py [hunk 2]
```diff
@@ -43,6 +43,13 @@
 INVALID_JSON = 'toto_est_pas_beau'
 
 
+# Python2/3
```
Comment (ltkum) [inline tests/integration/test_glstyles_storage.py]: why not import from chsdi/lib/helpers ?

File: chsdi/lib/files_handler.py [hunk 3]
```diff
@@ -218,7 +237,14 @@ def delete_file(self):
         }
 
     def _get_uuid(self):
-        return base64.urlsafe_b64encode(uuid.uuid4().bytes).replace('=', '')
+        # Python2/3
```
Comment (ltkum) [inline chsdi/lib/files_handler.py]: We have a TODO here.

File: chsdi/lib/sphinxapi/sphinxapi.py [hunk 4]
```diff
@@ -19,7 +19,28 @@
 import socket
 import re
 from struct import pack, unpack
+import six
+
+if six.PY3:
+    long = int
```
Comment (ltkum) [inline chsdi/lib/sphinxapi/sphinxapi.py]: I must say it at least once : overriding reserved words / types is something you should never do. ^_^

File: chsdi/lib/sphinxapi/sphinxapi.py [hunk 5]
```diff
@@ -936,6 +989,9 @@ def UpdateAttributes(self, index, attrs, values, mva=False):
         if mva:
             mva_attr = 1
         for attr in attrs:
+            # TODO Python2/3
```
Comment (ltkum) [inline chsdi/lib/sphinxapi/sphinxapi.py]: There is still a TODO here

**Pull request merged**